apps/oppsummeringsdokument-service/src/main/java/no/nav/registre/testnorge/oppsummeringsdokuemntservice/adapter/OppsummeringsdokumentAdapter.java

```diff
@@ -0,0 +1,184 @@
+                QueryBuilders.matchQuery("virksomheter.personer.arbeidsforhold.typeArbeidsforhold", value)
+        ));
+        return getAllCurrentDocumentsBy(
+                new NativeSearchQueryBuilder()
+                        .withQuery(combinedOnANDOperator(queryBuilders))
+                        .withPageable(pageable),
+                pageable
+        );
+    }
+
+
+    private List<Oppsummeringsdokument> getAllCurrentDocumentsBy(String miljo, LocalDate fom, LocalDate tom) {
+        var queryBuilders = new ArrayList<QueryBuilder>();
+        queryBuilders.add(QueryBuilders.matchQuery("miljo", miljo));
+        getKalendermaanedBetween(fom, tom).ifPresent(queryBuilders::add);
+
+        return getAllCurrentDocumentsBy(new NativeSearchQueryBuilder()
+                .withQuery(combinedOnANDOperator(queryBuilders))
+        );
+    }
+
+
+    private Optional<RangeQueryBuilder> getKalendermaanedBetween(LocalDate fom, LocalDate tom) {
+        if (fom == null && tom == null) {
+            return Optional.empty();
+        }
+        var builder = QueryBuilders.rangeQuery("kalendermaaned");
+
+        if (fom != null) {
+            builder.gte(fom.withDayOfMonth(1));
+        }
+
+        if (tom != null) {
+            builder.lte(tom.withDayOfMonth(tom.lengthOfMonth()));
+        }
+        return Optional.of(builder);
+    }
+
+    private QueryBuilder combinedOnANDOperator(List<QueryBuilder> list){
+        var queryBuilder = QueryBuilders.boolQuery();
+        for (var item : list){
+            queryBuilder.must(item);
+        }
+        return queryBuilder;
+    }
+
+    /**
+     * TODO Find a way to do this operation by elastic search
+     */
+    private List<Oppsummeringsdokument> getAllCurrentDocumentsBy(List<OppsummeringsdokumentModel> list) {
+        return list
+                .stream()
+                .collect(Collectors.groupingBy(s -> s.getKalendermaaned().withDayOfMonth(1) + s.getOpplysningspliktigOrganisajonsnummer()))
+                .values()
+                .stream()
+                .map(items -> items.stream().reduce(null, (total, value) -> {
+                    if (total == null || total.getVersion() < value.getVersion()) {
+                        total = value;
+                    }
+                    return total;
+                }))
+                .map(Oppsummeringsdokument::new)
+                .collect(Collectors.toList());
+    }
+
+
+    public Oppsummeringsdokument getCurrentDocumentBy(LocalDate kalendermaaned, String orgnummer, String miljo) {
+        var list = getAllCurrentDocumentsBy(
+                miljo,
+                kalendermaaned.withDayOfMonth(1),
+                kalendermaaned.withDayOfMonth(kalendermaaned.lengthOfMonth())
+        );
+
+        if (list.size() > 1) {
+            log.warn(
+                    "Fant flere en av samme versioner for kalendermaaned: {} og orgnummer: {}. Velger den første i listen.",
```

Comment:
Ville kanskje skrevet om denne til noe slikt i stedet: `"Fant flere med samme versjon for kalendermaaned:  {} og orgnummer: {}. Velger den først i listen."`